blog/2018/12/20/mojolicious-serving-angular-web-app/index.markdown

```diff
@@ -0,0 +1,379 @@
+    import { Component } from '@angular/core';
+    import { HttpClient } from '@angular/common/http';
+
+    @Component({
+        selector: 'app-root',
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+    export class AppComponent {
+        title = 'Mojolicious Angular web app';
+        adventDetail2018;
+        // Inject HttpClient into your component
+        constructor(private http: HttpClient) {}
+        ngOnInit(): void {
+            // Make the HTTP get request to mojolicious route
+            this.http.get('http://localhost:8080/advent/2018/detail').subscribe(data => {
+                this.adventDetail2018 = data;
+            });
+        }
+    }
+    Sachin@12:33 AM[~/workspace/project/mojo_angular/NgDemo/src/app]$ 
+
+##### c. Modify app.component.html file to show data fetched from backend Mojolicious
+Replace the default template(html) which was displaced earlier with our own html.
+I have just looped through `adventDetail2018` variable, which consists of data from http get request, using `*ngFor` built in directive to form a `table` body.
+
+    Sachin@12:34 AM[~/workspace/project/mojo_angular/NgDemo/src/app]$ cat app.component.html 
+    <div style="text-align:left">
+        <h1>
+            Welcome to {{ title }}!
+        </h1>
+        <h3>Mojolicious Advent Calendar 2018 Detail:</h3>
+        <table>
+            <thead>
+                <th>Day</th>
+                <th>Title</th>
+                <th>Author</th>
+            </thead>
+            <tbody>
+                <!--Angular two way data binding in action-->
+                <tr *ngFor="let d of adventDetail2018">
+                    <td>{{d.day}}</td>
+                    <td>{{d.title}}</td>
+                    <td>{{d.author}}</td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+    <router-outlet></router-outlet>
+
+##### d. app.component.css - Add little bit of style for the table. 
+Added a bare minimum style in `app.component.css` file to show its significance and how beautifully angular separates css away from html file.
+
+    Sachin@12:34 AM[~/workspace/project/mojo_angular/NgDemo/src/app]$ cat app.component.css 
+    table, th, td {
+        border: 1px solid black;
+        border-collapse: collapse;
+    }
+    th, td {
+        padding: 5px;
+        text-align: left;
+    }
+    Sachin@12:34 AM[~/workspace/project/mojo_angular/NgDemo/src/app]$
+
+##### e. Run `ng serve` command(as earlier) to see if angular app is looking good.
+
+##### f. Run `ng build --base-href=./` and copy `dist` folder content to mojolicious app's `public` directory as shown earlier
+
+##### 3. Run `hypnotoad` as shown earlier
+Visit `localhost:8080/NgDemo` in browser to witness wedding of Mojolicious and Angular:
+
+![final mojolicious serving angular SPA](final_mojo_angular_app.png)
+
+###### That's it. Lets revolutionize web app development. :)
```

Comment:
This is a grand claim for what is still a relatively simple demo. Can it be toned back to something like 

> That's all you need for simple single page app with angular and mojo. Take it further and see what is possible!

Also, again, not as a 6th level heading but as a paragraph

Reply:
Totally agree. :D